### PR TITLE
Seclathe Recipe Expansion

### PR DESCRIPTION
**Pull request by GoodWheatley**
## About the PR

 This PR adds a ton of recipes to the seclathe, allowing it to actually produce commonly used security gear, and guns. All the new recipes will have to be unlocked through sci before they can be produced. Additionally, at Emo's behest, I've allowed sci to print laser guns if they research the corresponding technologies.

IMPORTANT: I messed with the collision groups to get the xray cannon to work properly, if I've done it wrong please let me know.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Several new recipes for the seclathe have been added, that can be unlocked by research from Science.
- add: The protolathe can now print various types of laser weaponry once laser technologies are researched.

### CODE REVIEW

File: Content.Shared/Physics/CollisionGroup.cs [hunk 1]
```diff
@@ -14,14 +14,15 @@ namespace Content.Shared.Physics;
...
     None               = 0,
-    Opaque             = 1 << 0, // 1 Blocks light, can be hit by lasers
-    Impassable         = 1 << 1, // 2 Walls, objects impassable by any means
-    MidImpassable      = 1 << 2, // 4 Mobs, players, crabs, etc
-    HighImpassable     = 1 << 3, // 8 Things on top of tables and things that block tall/large mobs.
-    LowImpassable      = 1 << 4, // 16 For things that can fit under a table or squeeze under an airlock
-    GhostImpassable    = 1 << 5, // 32 Things impassible by ghosts/observers, ie blessed tiles or forcefields
-    BulletImpassable   = 1 << 6, // 64 Can be hit by bullets
-    InteractImpassable = 1 << 7, // 128 Blocks interaction/InRangeUnobstructed
+    XrayImpassable     = 1 << 0, // 1 Ignores walls, can only hit mobs
```
Comment (metalgearsloth) [inline Content.Shared/Physics/CollisionGroup.cs:17]: This should just be a mob layer, don't hardcode an entire layer for 1 interaction.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
A regular lathe should not be able to make 90% of the guns you added.

**Comment by EmoGarbage404** [comment]
emogarbage endorsed pr but i will come smite a lot of the design choices here once the tests work.

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml [hunk 2]
```diff
@@ -227,11 +227,15 @@
...
-      - PowerCellHigh  
+      - PowerCellHigh
       - SynthesizerInstrument
       - RPED
       - NodeScanner
       - HolofanProjector
+      - WeaponLaserCarbine
+      - WeaponAdvancedLaser
+      - WeaponLaserCannon
+      - WeaponXrayCannon
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Structures/Machines/lathe.yml:238]: These shouldn't be in the prototlathe. move them to secfab

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 3]
```diff
@@ -213,27 +213,218 @@
...
+    - Handcuffs
+    - Zipties
+    - StunBaton
+    - RiotShield
+    - RiotLaserShield
+    - RiotBulletShield
+    - Flash
+    - FlashPayload
+    - ForensicPad
+    - ForensicScanner
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:236]: why not put forensic stuff in a separate tech? this is a really big and bloated technology, especially for only 5000.

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 4]
```diff
@@ -213,27 +213,218 @@
...
+    - BasicSecurityTechnology
+  unlockedRecipes:
+    - MagazineBoxPistol
+    - MagazineBoxPistolRubber
+    - MagazineBoxMagnum
+    - MagazineBoxMagnumRubber
+    - MagazineBoxRifle
+    - MagazineBoxRifleRubber
+    - MagazineBoxLightRifle
+    - MagazineBoxLightRifleRubber
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:258]: printing entire boxes of ammunition seems pointless when you have magazines. Also, 5000 is too cheap.

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 5]
```diff
@@ -213,27 +213,218 @@
...
+- type: technology
+  name: technologies-ballistic-magazines
+  id: BallisticMagazinesTechnology
+  description: technologies-ballistic-magazines-description
+  icon:
+    sprite: Objects/Weapons/Guns/Ammunition/Magazine/Pistol/pistol_mag.rsi
+    state: red-icon
+  requiredPoints: 10000
+  requiredTechnologies:
+    - BallisticAmmunitionTechnology
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:269]: i don't get why you'd print whole boxes before just making magazines. Remove the boxes and just only have printable magazines.

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 6]
```diff
@@ -213,27 +213,218 @@
...
+  unlockedRecipes:
+    - SpeedLoaderMagnum
+    - SpeedLoaderMagnumHighVelocity
+    - SpeedLoaderMagnumRubber
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:273]: these could probably be their own thing.

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 7]
```diff
@@ -213,27 +213,218 @@
...
+  icon:
+    sprite: Objects/Weapons/Guns/SMGs/wt550.rsi
+    state: icon
+  requiredPoints: 20000
+  requiredTechnologies:
+    - BasicSecurityTechnology
+    - BallisticMagazinesTechnology
+  unlockedRecipes:
+    - WeaponRifleLecter
+    - WeaponSubMachineGunWt550
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:295]: you shouldn't be able to print rifles and smgs.

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 8]
```diff
@@ -213,27 +213,218 @@
...
+- type: technology
+  name: technologies-ballistic-weaponry
+  id: BallisticWeaponryTechnology
+  description: technologies-ballistic-weaponry-description
+  icon:
+    sprite: Objects/Weapons/Guns/SMGs/wt550.rsi
+    state: icon
+  requiredPoints: 20000
+  requiredTechnologies:
+    - BasicSecurityTechnology
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:291]: having this as a prereq isn't necessary because it's already a prereq for `BallisticMagazinesTechnology`

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 9]
```diff
@@ -213,27 +213,218 @@
...
+    sprite: Objects/Weapons/Guns/Battery/disabler.rsi
+    state: base
+  requiredPoints: 5000
+  requiredTechnologies:
+    - BasicSecurityTechnology
+  unlockedRecipes:
+    - Disabler
+    - Flashbang
+    - Bola
+    - FoodBoxDonut
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:314]: remove

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 10]
```diff
@@ -213,27 +213,218 @@
...
+  name: technologies-non-lethal
+  id: NonLethalTechnology
+  description: technologies-non-lethal-description
+  icon:
+    sprite: Objects/Weapons/Guns/Battery/disabler.rsi
+    state: base
+  requiredPoints: 5000
+  requiredTechnologies:
+    - BasicSecurityTechnology
+  unlockedRecipes:
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Research/technologies.yml:310]: all of the rubber bullets should be here instead of with the regular bullets.

File: Resources/Prototypes/Entities/Structures/Power/chargers.yml [hunk 11]
```diff
@@ -81,10 +82,10 @@
 - type: entity
   name: wall recharger
   id: WallWeaponCapacitorRecharger
-  parent: PowerCellRecharger
+  parent: WeaponCapacitorRecharger
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Structures/Power/chargers.yml:85]: why this change?

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
don't put regular bullets and high velocity in the same technology. They should be split up, because high velocity is more desirable.

File: Resources/Prototypes/Entities/Structures/Power/chargers.yml (hunk 11 above)
Comment (GoodWheatley) [reply Resources/Prototypes/Entities/Structures/Power/chargers.yml:85]: It didn't charge weapons like it was intended to

File: Resources/Prototypes/Catalog/Research/technologies.yml (hunk 9 above)
Comment (GoodWheatley) [reply Resources/Prototypes/Catalog/Research/technologies.yml:314]: remove the recipes or the nonlethal technology altogether?

Comment (EmoGarbage404) [reply Resources/Prototypes/Catalog/Research/technologies.yml]: remove the donut box

**Review by AJCM-git — CHANGES_REQUESTED** [review]
sloth review

**Comment by GoodWheatley** [comment]
closing this until I have the time to address reviews
